File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int ans = INT_MAX, sum = 0, s = 0;
        int sz = nums.size();

        for(int i=0; i<sz; i++){
            if(sum < target)
                sum += nums[i];
            if(sum >= target){
                while(sum >= target && s <= i){
                    ans = min(ans, i-s+1);
                    sum -= nums[s];
                    s++;
                }
            }
        }

        if(ans == INT_MAX) return 0;
        return ans;
    }
};

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int sz = nums.length, ans = sz+1, j=0, sum =0;

        for(int i=0; i<sz; i++){
            sum += nums[i];
            while(sum >= target && j <= i){
                ans = Math.min(ans, i-j+1);
                sum -= nums[j++];
            }
        }

        return ans == sz+1 ? 0 : ans;
    }
}